src/ddb/ddb.c

```diff
@@ -754,7 +754,12 @@ ddb_run_cmd(struct ddb_ctx *ctx, const char *cmd_str, bool write_mode)
 	struct argv_parsed	 parse_args = {0};
 	struct ddb_cmd_info	 info = {0};
 	int			 rc;
-	char			*cmd_copy = strdup(cmd_str);
+	char                    *cmd_copy;
+
+	D_STRNDUP(cmd_copy, cmd_str, strlen(cmd_str) + 1);
```

Comment:
I suspect this could cause a coverity issue.  Is there some maximum length that could be used instead?

Reply:
+1, I think this needs a true maximum

Reply:
fixed